content/blog/2019-07-02-grainchanger.md

```diff
@@ -0,0 +1,183 @@
+  ggtitle("winmove_agg")
+
+# standard aggregation using variance
+g_sf$dda <- nomove_agg(coarse_dat = g_sf,
+                         fine_dat = cont_ls, 
+                         agg_fun = var,
+                         quiet = TRUE)
+
+p4 <- ggplot(g_sf) + 
+  geom_sf(aes(fill = dda)) + 
+  scale_fill_viridis_c(name = "") +
+  theme_bw() + 
+  ggtitle("nomove_agg")
+
+(p1 + p2) / (p3 + p4)
+```
+
+![](img/blog-images/2019-07-02-grainchanger/example1-1.png)<!-- -->
+
+See the [package website](https://ropensci.github.io/grainchanger/articles/functions.html) for a description of the built-in functions. 
+
+### Why not just use the `raster` package?
+
+The key functionality of `grainchanger` is moving window smoothing, and aggregation to a coarser resolution. "But aren't these available in the `raster` package?", I hear you ask. 
+
+While `raster::focal` is super fast for calculating using `sum`, on large rasters it can become computationally cumbersome with other functions. However, using the weights matrix and the `sum` function, it is possible to calculate many other functions. Our package provides a wrapper to do so. 
+
+As a side note, it was a [helpful post from whuber on StackExchange](https://gis.stackexchange.com/questions/151962/calculating-shannons-diversity-using-moving-window-in-r) which led me to realise this.
+
+Additionally, the aggregation functions in the `grainchanger` package have been written so that (a) the user can aggregate to non-rectangular shapes; and (b) it can be run in parallel using the `future` package. This is particularly advantageous when using `winmove_agg`. Parallelisation can be achieved by adding the following line prior to running either `winmove_agg` or `nomove_agg`:
+
+```
+plan(multiprocess)
+```
+
+### Other features
+
+In testing the method, we needed to simulate landscapes (which we did with a great rOpenSci package [`NLMR`](https://ropensci.github.io/NLMR/)). To avoid edge effects, we wanted to create the effect of a torus (an infinite surface where the top joins the bottom, and the left side meets the right side), but could not find a function to do so - so we wrote one instead. Hopefully others will find this useful when working with simulated landscapes. 
+
+
+```r
+torus <- create_torus(cat_ls, 15)
+
+p1 <- show_landscape(cat_ls,
+                     legend_title = "",
+                     xlab = "", ylab = "") + 
+  ggtitle("Landscape")
+
+p2 <- show_landscape(torus,
+                     legend_title = "",
+                     xlab = "", ylab = "") + 
+  ggtitle("Landscape + torus")
+
+p1 + p2
+```
+
+![](img/blog-images/2019-07-02-grainchanger/example2-1.png)<!-- -->
+
+### Contributing
+
+We hope to continue to build on the functions so that they are as optimised as we can make them. If you have a particular function you would like to see included in the package, please get in touch, or submit a pull request making sure develop is the destination branch on the grainchanger repository.
```

Comment:
can you link "get in touch" (e.g. do you want them to open an issue?) and link to something that leads folks to the "develop" branch?